virtual_machines.go

```diff
@@ -152,23 +151,27 @@ func (v *VirtualMachine) Migrate(target, targetstorage string) (task *Task, err
 	return NewTask(upid, v.client), nil
 }
 
-func (v *VirtualMachine) Clone(name, target string) (newid int, task *Task, err error) {
+func (v *VirtualMachine) Clone(params *VirtualMachineCloneOptions) (newid int, task *Task, err error) {
```

Comment:
only thought here is newid is listed as required in the api... should we move that out of the Options and make it an int param which we allow passing in as 0 for a generated cluster.NextID? if you opt to refactor then add a comment saying pass 0 for the cluster's next ID

Reply:
I agree the client shouldn't deviate too much from the API design, but IMHO this is one of those times. Using your own ID management scheme is likely to be the edge case rather than the contrary, and thus should be kept to the Options struct rather than clutter every calls with `0`.

Reply:
i agree, let's go with this